domains/pallets/evm-tracker/src/lib.rs

```diff
@@ -79,12 +87,71 @@ mod pallet {
         #[pallet::weight(<T as frame_system::Config>::DbWeight::get().reads_writes(0, 1))]
         pub fn set_contract_creation_allowed_by(
             origin: OriginFor<T>,
-            contract_creation_allowed_by: PermissionedActionAllowedBy<T::AccountId>,
+            contract_creation_allowed_by: PermissionedActionAllowedBy<EthereumAccountId>,
         ) -> DispatchResult {
             ensure_root(origin)?;
             ContractCreationAllowedBy::<T>::put(contract_creation_allowed_by);
             Ok(())
         }
+
+        /// An inherent call to set ContractCreationAllowedBy.
+        #[pallet::call_index(1)]
+        #[pallet::weight((T::DbWeight::get().reads_writes(1, 1), DispatchClass::Mandatory))]
+        pub fn inherent_set_contract_creation_allowed_by(
+            origin: OriginFor<T>,
+            contract_creation_allowed_by: PermissionedActionAllowedBy<EthereumAccountId>,
+        ) -> DispatchResult {
+            ensure_none(origin)?;
+            ContractCreationAllowedBy::<T>::put(contract_creation_allowed_by);
+            Ok(())
+        }
+    }
+
+    #[pallet::inherent]
+    impl<T: Config> ProvideInherent for Pallet<T> {
+        type Call = Call<T>;
+        type Error = InherentError;
+        const INHERENT_IDENTIFIER: InherentIdentifier = INHERENT_IDENTIFIER;
+
+        fn create_inherent(data: &InherentData) -> Option<Self::Call> {
+            let inherent_data = data
+                .get_data::<InherentType>(&INHERENT_IDENTIFIER)
+                .expect("EVM tracker inherent data not correctly encoded")
+                .expect("EVM tracker inherent data must be provided");
+
+            inherent_data
+                .maybe_call
+                .map(T::IntoEvmTrackerCall::into_evm_tracker_call)
```

Comment:
Not sure we need this, as the `inherent_data` is the call arg it can be `inherent_data .map(|contract_creation_allowed_by| Call:: inherent_set_contract_creation_allowed_by { contract_creation_allowed_by })`